htdocs/main.php

```diff
@@ -174,14 +174,17 @@ function tplFromRequest($param)
     if (isset($caller->controlPanel)) {
         $tpl_data['control_panel'] = $caller->controlPanel;
     }
-
-    if (isset($caller->feedbackPanel) && $user->hasPermission('bvl_feedback')) {
-        $tpl_data['bvl_feedback']   = NDB_BVL_Feedback::bvlFeedbackPossible(
-            $TestName
-        );
-        $tpl_data['feedback_panel'] = $caller->feedbackPanel;
+    if (isset($caller->feedbackPanel)) {
+        if (!isset($user)) {
+            throw new Exception(401);
```

Comment:
Since the user is not logged in, the user object is initialized on a `NULL` value. Calling a method on a `NULL` value results into the page crashing. Not sure if this is the most appropriate place to catch the error, or in the user class itself.